### DISCUSSION TITLE
EF - Relacionamento N pra N com Fluent Mapping

### DISCUSSION THREAD

**Discussion by oimarcella** [balta-io/forum#596] [Tira Dúvidas] [answered]
Estou mapeando a relação N pra N de User e Role do projeto do curso de EF.

Vi que ele fica me retornando erro em users, pois espera que Role tenha uma lista de Users.
Porém se eu crio essa propriedade no model de role ficaria certo? 

Achei que nem sempre eu precisaria ter user dentro de role.  Pois user depende de role, mas o contrário não é verdade.
Eu estava fazendo isso, adicionando oque faltava no model de outras entidades, porém estou tentando entender se faz sentido.

![image](https://github.com/balta-io/forum/assets/43733159/33b11dba-d644-4973-b26f-915cc5ddaf7c)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @oimarcella, como vai?

Em uma relação N para N a dependência é mútua, senão seria uma relação 1 para N.
Então se os seu usuário pode ter muitos roles e um role pode ter muitos usuários, a relação é NxN.

Se o seu role pertence a seu usuário e seu usuário pode ter nenhum ou apenas um role, aí é uma relação 1xN!
